community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/commands/Predicate.scala

```diff
@@ -96,6 +96,53 @@ case class Ands(predicates: List[Predicate]) extends Predicate {
   override def atoms: Seq[Predicate] = predicates
 }
 
+case class AndedPropertyComparablePredicates(ident: Identifier, prop: Property, comparables: NonEmptyList[ComparablePredicate]) extends Predicate {
+
+  def symbolTableDependencies: Set[String] = comparables.foldLeft(prop.symbolTableDependencies) {
+    case (acc, predicate) => acc ++ predicate.symbolTableDependencies
+  }
+
+  def containsIsNull: Boolean = comparables.exists(_.containsIsNull)
+
+  def isMatch(m: ExecutionContext)(implicit state: QueryState): Option[Boolean] = {
```

Comment:
This is a complex function. Do you think it is possible to simplify it by using couple iterations over `comparables` or maybe with `foldLeft`?